classes/class.attachments.php

```diff
@@ -1003,27 +1003,19 @@ function get_field_types() {
             foreach ( $field_types as $type => $path ) {
                 // proceed with inclusion
                 if ( file_exists( $path ) ) {
+                    // Store a list of all declared classes before we include the file
+                    $existing_classes = get_declared_classes();
+                    
                     // include the file
                     include_once( $path );
 
-                    // store the registered classes so we can single out what gets added
-                    $existing_classes = get_declared_classes();
-
-                    // we're going to use our Attachments class as a reference because
-                    // during subsequent instantiations of Attachments (e.g. within template files)
-                    // these field classes WILL NOT be added to the array again because
-                    // we're using include_once() so that strategy is no longer useful
```

Comment:
This function is only called once during initialization and never again as it's return is cached in `$this->fields`  [in __construct](https://github.com/EdHurtig/attachments/blob/fix-field-type-loader-for-hhvm/classes/class.attachments.php#L71)  That variable should probably be checked by this function and returned on subsequent calls. If this function does that then it should also set `$this->fields` as well